### PR TITLE
Feature/dh 547 list interactions

### PR DESCRIPTION
**Pull request by ztolley**
Provides a top level link to list interactions, and a list of interactions using collection pattern.

No filters as back end doesn't support this yet.

screenshot from 2017-09-11 12-14-47

### CODE REVIEW

File: test/unit/apps/interactions/middleware.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      })
+    })
+    it('should call the search response transformer with the interaction entity transformer', async () => {
+      await this.middleware.getInteractionsCollection(this.req, this.res, this.next)
+
+      expect(this.transformApiResponseToSearchCollectionStub).to.be.calledWith(
+        { query: this.req.query },
+        transformInteractionToListItem
+      )
+    })
```
Comment (yeahfro) [inline test/unit/apps/interactions/middleware.test.js]: I see what is happening here. My feelings around this style of tests is that it is testing implementation details which can be costly if the code ever changes. My preferred approach would be to do similar to what you have done with setting up the mocks and then just asserting what comes back out.. I cannot see any assertions on `res.locals` and that is what we actually care about on this controller function.

Comment (ztolley) [reply test/unit/apps/interactions/middleware.test.js]: What I was worried about was duplication of testing and would end up re-testing the item transformer, but I'll just mock that out and test the result

File: src/apps/interactions/router.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
 const router = require('express').Router()
 
+const { setDefaultQuery } = require('../middleware')
+const { getInteractionsCollection, getRequestBody } = require('./middleware')
+const { renderInteractionList } = require('./controllers/list')
```
Comment (feedmypixel) [inline src/apps/interactions/router.js]: can these controllers can be merged?

Comment (ztolley) [reply src/apps/interactions/router.js]: ??

Comment (feedmypixel) [reply src/apps/interactions/router.js]: you can use `const { detailsController, listController, editController } = require('./controllers')`

Comment (ztolley) [reply src/apps/interactions/router.js]: One thing that bugs me is some places we use module.method and other places we just use method

**Pull request merged**